File: ios/Classes/crashlytics/CrashlyticsBridge.swift
import OSAMCommon

class CrashlyticsBridge: NSObject, FlutterStreamHandler, CrashlyticsWrapper {
    private var eventSink: FlutterEventSink?
    
    func recordException(className: String, stackTrace: String) {
        eventSink?(["className": className, "stackTrace": stackTrace])
    }
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}

File: android/src/main/kotlin/cat/bcn/commonmodule/flutter/osam_common_module_flutter/crashlytics/CrashlyticsBridge.kt
package cat.bcn.commonmodule.flutter.osam_common_module_flutter.crashlytics

import cat.bcn.commonmodule.crashlytics.CrashlyticsWrapper
import io.flutter.plugin.common.EventChannel

class CrashlyticsBridge : EventChannel.StreamHandler, CrashlyticsWrapper {

    private var eventSink: EventChannel.EventSink? = null

    override fun recordException(exception: Exception) {
        eventSink?.success(
            mapOf(
                "className" to exception::class.qualifiedName,
                "stackTrace" to exception.stackTraceToString()
            )
        )
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}